docs/pairing-architecture.md

```diff
@@ -0,0 +1,292 @@
+## Pairing Flow Architecture
+
+To allow users to quickly connect a mobile browser to Firefox Sync, without having to type their password,
+by pairing with an authenticated Firefox profile on their PC via scanning a QR code.
+See the [Project Plan Document](https://docs.google.com/document/d/1mFf0sfEK8o1csXLeyK1x4GS9SUMIq4q8bu25LiydPew/edit#) for more information.
+
+The Desktop and Mobile devices will not communicate directly, because we cannot assume that they will be on
+the same network and because this is difficult to achieve from web content.
+Instead, they will exchange encrypted messages via websocket to a lightweight "channel server" service which can relay messages back and forth:
+
```

Comment:
This intro could be expanded upon to include info about what information the new device needs and how that info is received from the client.

In essence we are doing a remote authorization where the already authenticated device generates Sync keys and an OAuth code for the new device. PKCE facilitates the transfer of the Sync key between devices and the channel server facilitates direct communication between the two, including the final transfer of the OAuth code.

The security properties section from the feature doc would also be useful.

Reply:
"remote authorization" is a great way to describe this!

Reply:
👍